### DISCUSSION TITLE
Trouble with coordinates

### DISCUSSION THREAD

**Discussion by elisamist** [opengisch/QField#5798] [Q&A]
Hi! 
So I am very new to using QGis and QField, but I have managed to get my project in QField and sync with Qgis. So my project has a few layers and I would like to maybe add some points to said layers out in the field. However, when I have been doing some test points, they never include the x,y,z coordinates, how would I go about making QField automatically fill in the coordinates so when I sync I can see them in QGis. The Fields I have made for the layers are fine and set correctly to recieve the data but its just not working properly.

Thank you in advance, and I hope you understand my problem

**Comment by beanzmo** [comment]
@elisamist, can you elaborate a bit more. 
Are you trying to add the x,y,z coordinates to your attribute table 

If that is the case you will have to 

1. Add three new attributes called Longitude and Latidude
2. Under 'properties' -> attribute form (change to second option - Drag and Drop Designer)
3. Remove the x, y, z from your attribute form -> in the 'available widgets' field direct to 
4. x coordinate and add x( $geometry ) to the 'Defaults tab' 
![image](https://github.com/user-attachments/assets/1ad21407-75d5-4e49-baad-05be7ce1b341)
5. Do the same for y and z

Hopefully, I understood your question correctly. If that does not work let me know and please share your project if it is not too big.

**Reply by elisamist** [reply]
Hi! and thank you for answering. Yes, so I have actually done this for the x, y and z fields I have made. But what I am wondering is that when I am testing it out in QField on my tablet and I make a new measurement, and then sync it and look at the attribute table on my desktop all the fields show a NULL value
![Skjámynd 2024-11-13 113050](https://github.com/user-attachments/assets/85446517-5495-4a03-a01c-2316bed3c491)
like on the picture, the first two points I made I had to manually add the coordinates with the field calculator, and the newest point(Test3) which has the attribute for just like you show just shows NULL  values. I am just wondering if there is a way to make this more automated, so when I sync the new measurements that the coordinates are automatically there when i look at the attribute table on the desktop. 

Best.
